sentry-android-ndk/src/main/jni/sentry.c

```diff
@@ -11,8 +11,132 @@ struct transport_options {
+        jstring id,
+        jstring email,
+        jstring ipAddress,
+        jstring username) {
+    sentry_value_t user = sentry_value_new_object();
+    if (id) {
+        const char *charId = (*env)->GetStringUTFChars(env, id, 0);
+        sentry_value_set_by_key(user, "id", sentry_value_new_string(charId));
+    }
+    if (email) {
+        const char *charEmail = (*env)->GetStringUTFChars(env, email, 0);
+        sentry_value_set_by_key(
+                user, "email", sentry_value_new_string(charEmail));
+    }
+    if (ipAddress) {
+        const char *charIpAddress = (*env)->GetStringUTFChars(env, ipAddress, 0);
+        sentry_value_set_by_key(
+                user, "ip_address", sentry_value_new_string(charIpAddress));
+    }
+    if (username) {
+        const char *charUsername = (*env)->GetStringUTFChars(env, username, 0);
+        sentry_value_set_by_key(
+                user, "username", sentry_value_new_string(charUsername));
+    }
+    sentry_set_user(user);
+}
+
+JNIEXPORT void JNICALL Java_io_sentry_android_ndk_NativeScope_nativeRemoveUser(
+        JNIEnv *env,
+        jclass cls) {
+    sentry_remove_user();
+}
+
+JNIEXPORT void JNICALL Java_io_sentry_android_ndk_NativeScope_nativeAddBreadcrumb(
+        JNIEnv *env,
+        jclass cls,
+        jstring level,
+        jstring message,
+        jstring category,
+        jstring type,
+        jstring timestamp,
+        jstring data) {
+    if (!level && !message && !category && !type) {
+        return;
+    }
+    const char *charMessage = NULL;
+    if (message) {
+        charMessage = (*env)->GetStringUTFChars(env, message, 0);
+    }
+    const char *charType = NULL;
+    if (type) {
+        charType = (*env)->GetStringUTFChars(env, type, 0);
+    }
+    sentry_value_t crumb
+            = sentry_value_new_breadcrumb(charType, charMessage);
+
+    if (category) {
+        const char *charCategory = (*env)->GetStringUTFChars(env, category, 0);
+        sentry_value_set_by_key(
+                crumb, "category", sentry_value_new_string(charCategory));
+    }
+    if (level) {
+        const char *charLevel = (*env)->GetStringUTFChars(env, level, 0);
+        sentry_value_set_by_key(
+                crumb, "level", sentry_value_new_string(charLevel));
+    }
+
+    if (timestamp) {
+        // overwrite timestamp that is already created on sentry_value_new_breadcrumb
+        const char *charTimestamp = (*env)->GetStringUTFChars(env, timestamp, 0);
+        sentry_value_set_by_key(
+                crumb, "timestamp", sentry_value_new_string(charTimestamp));
+    }
+
+    if (data) {
+        const char *charData = (*env)->GetStringUTFChars(env, data, 0);
+
+        // we create an object because the Java layer parses it as a Map
+        sentry_value_t dataObject = sentry_value_new_object();
+        sentry_value_set_by_key(dataObject, "data", sentry_value_new_string(charData));
+
+        sentry_value_set_by_key(crumb, "data", dataObject);
```

Comment:
this could be improved if we expose something like `sentry_value_from_json`
cc @Swatinem